pkgs/applications/office/zotero-dev/default.nix

```
@@ -0,0 +1,400 @@
+        mkdir $out
+        cp -r . $out
+      '';
+    };
+
+  zotero-build = stdenv.mkDerivation {
+    pname = "${pname}-build";
+    inherit version meta;
+    src = fetchFromGitHub {
+      owner = "zotero";
+      repo = "zotero-build";
+      rev = "00e854c6588f329b714250e450f4f7f663aa0222";
+      hash = "sha256-Gvt37jObgSQ10GBYjnCLu5XbUAy3oVTkWPvHbhLF+fw=";
+      fetchSubmodules = true;
+    };
+
+    postPatch = ''
+      sed -i -E "/-aL/a '--chmod=Du=rwx'," xpi/build_xpi
+      #sed -i 's/-aL/-L/' xpi/build_xpi
+    '';
+
+    buildInputs = [ python3 rsync perl ];
+
+    buildPhase = ''
+      python3 xpi/build_xpi -s ${zotero-client}/build -c source -m ${rev}
+    '';
+
+    installPhase = ''
+      mkdir $out
+      cp -r . $out
+    '';
+  };
+
+in
+
+stdenv.mkDerivation {
+  inherit pname version meta;
+
+  src = fetchFromGitHub {
+    owner = "zotero";
+    repo = "zotero-standalone-build";
+    rev = "e9ef6bf21d39cc66f1edefdd5b7429bbaf0c5247";
+    hash = "sha256-NcnbqCN6Pti2KuVX79QLrTk2V/3sMxMrbgRzSTluOtM=";
+    fetchSubmodules = true;
+  };
+
+  patches = [ ./fetchxul.patch ];
+
+  postPatch = ''
+    patchShebangs .
+
+    sed -i 's|LINUX_i686_RUNTIME_PATH=.*|LINUX_i686_RUNTIME_PATH="$DIR/xulrunner/firefox"|' config.sh
+    sed -i 's|LINUX_x86_64_RUNTIME_PATH=.*|LINUX_x86_64_RUNTIME_PATH="$DIR/xulrunner/firefox"|' config.sh
+    sed -i 's|ZOTERO_SOURCE_DIR=.*|ZOTERO_SOURCE_DIR="${zotero-client}"|' config.sh
+    sed -i 's|ZOTERO_BUILD_DIR=.*|ZOTERO_BUILD_DIR="${zotero-build}"|' config.sh
+
+    sed -i -E 's|(.*hash=).*|\1${rev}|' scripts/dir_build
+    sed -i '/build_xpi/d' scripts/dir_build
+
+    sed -i -E 's|(rsync -a.*)|\1; chmod -R +w $BUILD_DIR|' build.sh
+    #sed -i 's|MaxVersion=.*|MaxVersion=111.0|' assets/application.ini
+  '';
+
+  nativeBuildInputs = [ makeWrapper python3 unzip zip perl rsync wrapGAppsHook ];
+  buildInputs = [ gsettings-desktop-schemas glib gtk3 gnome.adwaita-icon-theme dconf ];
+
+  configurePhase = ''
+    mkdir xulrunner
+    cp -Lr ${firefox-esr-102-unwrapped}/lib/firefox xulrunner
+    chmod -R +w xulrunner
+
+    cp -Lr ${pdftools} pdftools
+    chmod -R +w pdftools
+
+    ./fetch_xulrunner.sh -p l
+  '';
+
+  buildPhase = ''
+    chmod -R +w /build
+    scripts/dir_build -p l
+  '';
+
+  installPhase =
+    let
+      desktopItem = makeDesktopItem {
+        name = pname;
+        exec = "${pname} -url %U";
+        icon = "zotero";
+        comment = meta.description;
+        desktopName = "Zotero";
+        genericName = "Reference Management";
+        categories = [ "Office" "Database" ];
+        startupNotify = true;
+        startupWMClass = pname;
+        mimeTypes = [ "x-scheme-handler/zotero" "text/plain" ];
+        terminal = false;
+        actions = {
+          profile-manager-window = {
+            name = "Profile Manager";
+            exec = "${pname} -P";
+          };
+        };
+      };
+    in
+    ''
+      mkdir -p $out/bin
+      mkdir -p $out/usr/lib
```

```suggestion
      mkdir -p $out/bin $out/usr/lib
```
also we don't really use usr in nixpkgs, $out/lib would be better